File: stockify_app_flutter/lib/feature/user/service/user_service.dart
import 'package:stockify_app_flutter/feature/user/model/user.dart';
import 'package:stockify_app_flutter/feature/user/model/user_filter_param.dart';

abstract class UserService {
  void addUser(User user);

  void editUser(User user);

  User getUser(int id);

  List<User> getAllUsers();

  void deleteUser(int id);

  List<User> getFilteredUsers(UserFilterParams params);
}

File: stockify_backend_golang/src/feature/user/service/user_service.go
package service

import "stockify_backend_golang/src/feature/user/model"

type UserService interface {
	AddUser(item model.User)
	GetAllUsers() []model.User
	GetUserById(id uint64) model.User
	UpdateUser(User model.User)
	DeleteUserById(id uint64)
	GetFilteredUsers(params model.UserQueryParams) ([]model.User, error)
}
